contracts/helpers.cairo

```diff
@@ -381,3 +395,62 @@ func fromUint256_balance{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_c
     }
     return x__;
 }
+
+
+// equivalent to toUint256_balance, but for option_position, which is stored as a felt (Int),
+// because we don't need the full range of uint256
+func toInt_balance{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    x: Math64x61_,
+    currency_address: Address
+) -> Int {
+    let (decimal) = get_decimal(currency_address);
+    let (dec_) = pow10(decimal);
+    let x_ = x * dec_;
+
+    with_attr error_message("x_ out of bounds in toUint256_balance"){
+        assert_le(x, Math64x61.BOUND);
+        assert_le(-Math64x61.BOUND, x);
+    }
+    let amount_felt = Math64x61.toFelt(x_);
+    assert_nn(amount_felt);
+
+    return amount_felt;
+}
+
+func fromInt_balance{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    x: Int,
+    currency_address: Address
+) -> Math64x61_ {
+    let (decimal) = get_decimal(currency_address);
+    let (dec_) = pow10(decimal);
+
+
+    let x_ = Math64x61.fromFelt(x);
+    with_attr error_message("fromInt_balance failed signed_div_rem( {dec_} {x__} ") {
+        let (x__, _) = signed_div_rem(x_, dec_, Math64x61.BOUND);
```

Comment:
@tensojka btw, math64x61 also uses this `Math64x61.assert64x61(x__);` to make sure the result is not out of bounds

Reply:
not sure what you mean, do you want to custom fn to check that the x is < 2**64?

Reply:
m64x61 already does this in fromFelt

Reply:
and Math64x61.BOUND does the same thing... I'm actually wondering why it is implemented in Math64x61 div...